src/object_detection.cpp

```diff
@@ -56,50 +68,75 @@ class Image_Finder {
             return;
         }
 
-        // Save the OpenCV image to a file
-        std::string image_filename = "/home/david/Documents/catkin_ws/src/object_detection_pkg/temp_files/image.png";  // Replace with your desired file path
-        if (cv::imwrite(image_filename, cv_ptr->image)) {
-            ROS_INFO("Saved image to %s", image_filename.c_str());
-
-            // Run the Python script and capture its output
-            std::string python_command = "python3 /home/david/Documents/catkin_ws/src/object_detection_pkg/src/yolov5_image_analyzer.py " + image_filename;
-            FILE* pipe = popen(python_command.c_str(), "r");
-            if (!pipe) {
-                ROS_ERROR("Failed to execute the Python script.");
-                return;
-            }
-
-            char buffer[128];
-            std::string script_output = "";
-            while (fgets(buffer, sizeof(buffer), pipe) != NULL) {
-                script_output += buffer;
-            }
-            pclose(pipe);
-
-            // TODO Parse the JSON output here (script_output variable contains the JSON data)
-            //parseAndProcessJSONOutput(script_output);
-        } else {
+        // Saving the OpenCV image to a file
+        std::string image_filename =  package_path + "/temp_files/image.png";
```

Comment:
Good idea to use instead on environment variables! Just make sure you add temp_files/* to the gitignore for the repo!

Reply:
Great idea! Done!